.github/workflows/push_pr.yml

```diff
@@ -25,7 +25,14 @@ jobs:
   compile_and_test:
     strategy:
       matrix:
-        flink: [1.16-SNAPSHOT, 1.17-SNAPSHOT, 1.18-SNAPSHOT, 1.19-SNAPSHOT]
+        flink: [ 1.16-SNAPSHOT, 1.17-SNAPSHOT ]
+        jdk: [ '8, 11' ]
+        include:
+          - flink: 1.18-SNAPSHOT
+            jdk: '8, 11, 17'
+          - flink: 1.19-SNAPSHOT
+            jdk: '8, 11, 17, 21'
```

Comment:
I'm not sure current matrix is you want or not, maybe you need something like if I understand azure pipeline yaml correctly?
```
        flink: [1.16-SNAPSHOT, 1.17-SNAPSHOT, 1.18-SNAPSHOT, 1.19-SNAPSHOT]
        jdk: ['8, 11, 17, 21']
        exclude:
          - flink: 1.16-SNAPSHOT
            jdk: '17, 21'
          - flink: 1.17-SNAPSHOT
            jdk: '21'
```

Reply:
This modification refers to the jdbc-connector.

https://github.com/apache/flink-connector-jdbc/blob/e3dd84160cd665ae17672da8b6e742e61a72a32d/.github/workflows/push_pr.yml#L19-L38